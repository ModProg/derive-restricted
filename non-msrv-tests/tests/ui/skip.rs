use std::marker::PhantomData;

use derive_where::DeriveWhere;

#[derive(DeriveWhere)]
#[derive_where(Debug)]
struct DuplicateSkipSame<T>(#[derive_where(skip, skip)] PhantomData<T>);

#[derive(DeriveWhere)]
#[derive_where(Debug)]
struct DuplicateSkipSeparate<T>(
	#[derive_where(skip)]
	#[derive_where(skip)]
	PhantomData<T>,
);

#[derive(DeriveWhere)]
#[derive_where(Debug)]
struct EmptySkip<T>(#[derive_where(skip())] PhantomData<T>);

#[derive(DeriveWhere)]
#[derive_where(Debug)]
struct OverridingSkip<T>(
	#[derive_where(skip)]
	#[derive_where(skip(Debug))]
	PhantomData<T>,
);

#[derive(DeriveWhere)]
#[derive_where(Debug)]
struct UnderridingSkip<T>(
	#[derive_where(skip(Debug))]
	#[derive_where(skip)]
	PhantomData<T>,
);

#[derive(DeriveWhere)]
#[derive_where(Clone; T)]
struct NoSupportedTrait<T>(#[derive_where(skip)] PhantomData<T>);

#[derive(DeriveWhere)]
#[derive_where(Clone; T)]
struct UnsupportedTrait<T>(#[derive_where(skip(Clone))] PhantomData<T>);

#[derive(DeriveWhere)]
#[derive_where(Debug)]
struct DuplicateTraitSame<T>(#[derive_where(skip(Debug, Debug))] PhantomData<T>);

#[derive(DeriveWhere)]
#[derive_where(Debug)]
struct DuplicateTraitSeparate<T>(
	#[derive_where(skip(Debug))]
	#[derive_where(skip(Debug))]
	PhantomData<T>,
);

#[derive(DeriveWhere)]
#[derive_where(Clone; T)]
struct MissingDeriveTrait<T>(#[derive_where(skip(Debug))] PhantomData<T>);

fn main() {}
