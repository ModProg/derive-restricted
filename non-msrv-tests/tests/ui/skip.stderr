error: duplicate `skip` option
 --> tests/ui/skip.rs:4:50
  |
4 | struct DuplicateSkipSame<T>(#[derive_where(skip, skip)] PhantomData<T>);
  |                                                  ^^^^

error: duplicate `skip` option
 --> tests/ui/skip.rs:9:17
  |
9 |     #[derive_where(skip)]
  |                    ^^^^

error: empty attribute option found
  --> tests/ui/skip.rs:14:36
   |
14 | struct EmptySkip<T>(#[derive_where(skip())] PhantomData<T>);
   |                                    ^^^^^^

error: unexpected constraint on `skip` when unconstrained `skip` already used
  --> tests/ui/skip.rs:19:17
   |
19 |     #[derive_where(skip(Debug))]
   |                    ^^^^^^^^^^^

error: duplicate `skip` option
  --> tests/ui/skip.rs:26:17
   |
26 |     #[derive_where(skip)]
   |                    ^^^^

error: no trait that can be skipped is being implemented
  --> tests/ui/skip.rs:31:43
   |
31 | struct NoSupportedTrait<T>(#[derive_where(skip)] PhantomData<T>);
   |                                           ^^^^

error: unsupported `Clone` constraint on `skip`
  --> tests/ui/skip.rs:34:48
   |
34 | struct UnsupportedTrait<T>(#[derive_where(skip(Clone))] PhantomData<T>);
   |                                                ^^^^^

error: duplicate `Debug` constraint on `skip`
  --> tests/ui/skip.rs:37:57
   |
37 | struct DuplicateTraitSame<T>(#[derive_where(skip(Debug, Debug))] PhantomData<T>);
   |                                                         ^^^^^

error: duplicate `Debug` constraint on `skip`
  --> tests/ui/skip.rs:42:22
   |
42 |     #[derive_where(skip(Debug))]
   |                         ^^^^^

error: trait to be skipped isn't being implemented
  --> tests/ui/skip.rs:47:50
   |
47 | struct MissingDeriveTrait<T>(#[derive_where(skip(Debug))] PhantomData<T>);
   |                                                  ^^^^^
